azkaban-common/src/main/java/azkaban/utils/OsMemoryUtil.java

```diff
@@ -0,0 +1,110 @@
+package azkaban.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Utility class for getting system memory information
+ */
+class OsMemoryUtil {
+  private static final Logger logger = LoggerFactory.getLogger(SystemMemoryInfo.class);
+
+  static final String MEMINFO_FILE = "/proc/meminfo";
+
+  private enum MemKey {MemFree, Buffers, Cached, SwapFree}
+
+  /**
+   *
+   * @return true if the meminfo file exists.
+   */
+  boolean doesMemInfoFileExist() {
+    File f = new File(MEMINFO_FILE);
+    return f.exists() && !f.isDirectory();
+  }
+
+  /**
+   * Includes OS cache and free swap.
+   * @return the total free memory size of the OS. 0 if there is an error.
+   */
+  long getOsTotalFreeMemorySize() {
+    List<String> lines;
+    // The file /proc/meminfo seems to contain only ASCII characters.
+    // The assumption is that the file is not too big. So it is simpler to read the whole file into memory.
+    try {
+      lines = Files.readAllLines(Paths.get(MEMINFO_FILE), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      String errMsg = "Failed to open mem info file: " + MEMINFO_FILE;
+      logger.warn(errMsg, e);
+      return 0;
+    }
+    return getOsTotalFreeMemorySizeFromStrings(lines);
+  }
+
+  /**
+   *
+   * @param lines text lines from the procinfo file
+   * @return the total size of free memory in kB. 0 if there is an error.
+   */
+  long getOsTotalFreeMemorySizeFromStrings(List<String> lines) {
+    long totalFree = 0;
+    int count = 0;
+
+    MemKey[] keys = MemKey.values();
+    for (String line : lines) {
+      for (MemKey key : keys) {
+        String keyName = key.name();
+        if (line.startsWith(keyName)) {
+          count++;
+          long size = parseMemoryLine(line);
+          if (size == 0) {
+            return 0;
+          }
+          totalFree += size;
+        }
+      }
+    }
+
+    int length = keys.length;
+    if (count != length) {
+      String errMsg = String.format("Expect %d keys in the meminfo file. Got %d. content: %s", length, count, lines);
+      logger.warn(errMsg);
+      totalFree = 0;
+    }
+    return totalFree;
+  }
+
+  /**
+   * Example file:
+   * $ cat /proc/meminfo
+   MemTotal:       65894008 kB
+   MemFree:        59400536 kB
+   Buffers:          409348 kB
+   Cached:          4290236 kB
+   SwapCached:            0 kB
+
+   * Make the method package private to make unit testing easier.
+   * Otherwise it can be made private.
+
+   * @param line the text for a memory usage statistics we are interested in
+   * @return size of the memory. unit kB. 0 if there is an error.
+   */
+  long parseMemoryLine(String line) {
```

Comment:
Yes, same here. I did some research on the same.

Turns out that there is no platform independent way to do so.